From 06de2b1f1b1de06cda47ffac6577794447e7562b Mon Sep 17 00:00:00 2001
From: Thales Elero Cervi <thaleselero.cervi@windriver.com>
Date: Mon, 19 Dec 2022 17:19:34 -0300
Subject: [PATCH] Update libvirt configuration script for Debian

Update the libvirt.sh template including a conditional check for the
Linux distribution before setting the kvm device permission and
ownership.
This was necessary when the libvirt image was stx-centos based but
now that it is stx-debian based, the newest package versions for both
libvirt and qemu rely on Debian systemd [1] to manage the it.

[1] https://salsa.debian.org/systemd-team/systemd/-/commit/4fc3fa53bfa6e16ceb6cd312f49003839b56144a

Signed-off-by: Thales Elero Cervi <thaleselero.cervi@windriver.com>
Change-Id: Ib2c3b0cbd0666ffd5dd310103c3f1a75b36c330c
[ Adapt openstack-helm-infra patches to Epoxy release ]
Signed-off-by: Daniel Caires <DanielMarques.Caires@windriver.com>
---
 libvirt/templates/bin/_libvirt.sh.tpl | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libvirt/templates/bin/_libvirt.sh.tpl b/libvirt/templates/bin/_libvirt.sh.tpl
index 81813f4f5..af1b4f5e0 100644
--- a/libvirt/templates/bin/_libvirt.sh.tpl
+++ b/libvirt/templates/bin/_libvirt.sh.tpl
@@ -41,9 +41,11 @@ fi
 
 rm -f /var/run/libvirtd.pid
 
-if [[ -c /dev/kvm ]]; then
+if [ "$(cat /etc/os-release | grep -w NAME= | grep -w CentOS)" ]; then
+  if [[ -c /dev/kvm ]]; then
     chmod 660 /dev/kvm
     chown root:kvm /dev/kvm
+  fi
 fi
 
 #Setup Cgroups to use when breaking out of Kubernetes defined groups
-- 
2.34.1
